cvat/apps/engine/background.py

```diff
@@ -601,11 +630,12 @@ def export(self) -> Response:
         )
         rq_job = queue.fetch_job(rq_id)
 
-        with suppress(ReenqueueJobException):
+        # ensure that there is no race condition when processing parallel requests
+        with get_rq_lock_by_job(rq_job), suppress(ReenqueueJobException):
```

Comment:
Isn't it better to acquire the lock before the job is fetched? I think it's not guaranteed that job data in the fetched object is actual after the lock is acquired.

Reply:
I missed that, thanks